program/tests/replenish.rs

```diff
@@ -0,0 +1,431 @@
+#![allow(clippy::arithmetic_side_effects)]
+#![cfg(feature = "test-sbf")]
+
+mod helpers;
+
+use {
+    helpers::*,
+    solana_program_test::*,
+    solana_sdk::{
+        account::AccountSharedData,
+        pubkey::Pubkey,
+        signature::Signer,
+        stake::{
+            stake_flags::StakeFlags,
+            state::{Delegation, Stake, StakeStateV2},
+        },
+        sysvar::{clock::Clock, stake_history::StakeHistory},
+        transaction::Transaction,
+    },
+    spl_single_pool::{error::SinglePoolError, id, instruction},
+    test_case::test_case,
+};
+
+async fn replenish(context: &mut ProgramTestContext, vote_account: &Pubkey) {
+    let instruction = instruction::replenish_pool(&id(), vote_account);
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+
+    refresh_blockhash(context).await;
+}
+
+#[test_case(false, false; "noop")]
+#[test_case(true, false; "pool")]
+#[test_case(false, true; "onramp")]
+#[test_case(true, true; "pool_precedence")]
+#[tokio::test]
+async fn reactivate_success(reactivate_pool: bool, fund_onramp: bool) {
```

Comment:
nit naming?
```suggestion
async fn replenish_success(replenish_pool: bool, fund_onramp: bool) {
```

Reply:
oh, i named this one that because it tests the reactivation logic in isolation, and the value moving logic is tested in `move_value_success()`

Reply:
Ah gotcha, makes sense!